mesonbuild/compilers/fortran.py

```diff
@@ -182,20 +182,28 @@ def __init__(self, exelist, version, for_machine: MachineChoice,
 
     def get_options(self):
         opts = FortranCompiler.get_options(self)
-        fortran_stds = ['legacy', 'f95', 'f2003']
-        if version_compare(self.version, '>=4.4.0'):
-            fortran_stds += ['f2008']
-        if version_compare(self.version, '>=8.0.0'):
-            fortran_stds += ['f2018']
+        fortran_stds = ['legacy', 'f95', 'f2003', 'f2008', 'f2018']
         opts.update({'fortran_std': coredata.UserComboOption('Fortran language standard to use',
                                                              ['none'] + fortran_stds,
                                                              'none')})
         return opts
 
     def get_option_compile_args(self, options) -> typing.List[str]:
+        """
+        While many C/C++ compilers default to an old standard that doesn't work with modern syntax,
+        Fortran compilers default to the highest possible language standard.
+        Thus it's better *not* to enforce a standard the compiler isn't fully capable of.
+        This is normal practice, including with CMake.
+        """
         args = []
         std = options['fortran_std']
         if std.value != 'none':
+            if std.value == 'f2018' and version_compare(self.version, '<8.0.0'):
+                # Gfortran < 8.0 didn't have f2018
+                return args
+            if std.value == 'f2008' and version_compare(self.version, '<4.4.0'):
+                # Gfortran < 4.4 didn't have f2018
```

Comment:
There seems to be some mismatch here. Is it 2018 or 2008?

And further, this seems to change the behaviour IIRC. It would show 2018 as a selectable option even for older compilers that don't support it.

Reply:
This is a comment typo, the line 205 comment should say f2008